http2-flowcontrol-sim/src/Connection.java

```diff
@@ -0,0 +1,119 @@
+import java.util.ArrayList;
+import java.util.HashMap;
+
+public class Connection extends SimulationListener implements ClockListener {
+
+  private HashMap<Integer, Stream> streams;
+  private int initialWindow;
+  private int availableWindow;
+  private int publishedWindow;
+  private SimulationClock controller;
+  private int bandwidthAvailable;
+  private int latency;
+  private int bandwidth;
+
+  public Connection(SimulationClock clock, int initWindow, int latency, int bandwidth) {
+    streams = new HashMap<Integer, Stream>();
```

Comment:
All the things that are not dependent on the input params should be moved up and out of the constructor.